pkgs/misc/cups/drivers/cups-pdf-to-pdf/default.nix

```
@@ -0,0 +1,64 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cups
+, makeBinaryWrapper
+, coreutils
+, nixosTests
+}:
+
+stdenv.mkDerivation rec {
+  pname = "cups-pdf-to-pdf";
+  version = "unstable-2021-12-22";
+  src = fetchFromGitHub {
+    owner = "alexivkin";
+    repo = "CUPS-PDF-to-PDF";
+    rev = "c14428c2ca8e95371daad7db6d11c84046b1a2d4";
+    hash = "sha512-FP+GGq3hVAXrLzidAsp3Q1WNzZxrIDaQHN36+RTu1W09FlOfZnIVdw/uJHBMWSm03HIBPSwQmpmpW8KqRv6izQ==";
+  };
+  buildInputs = [ cups ];
+  # cups-pdf wants to run with root privileges;
+  # to facilitate the use of a suid root wrapper,
+  # we have to use the binary wrapper here
+  nativeBuildInputs = [ makeBinaryWrapper ];
+  # gcc command line is taken from original cups-pdf's README file
+  # https://fossies.org/linux/cups-pdf/README
+  buildPhase = ''
+    runHook preBuild
+    gcc -O9 -s cups-pdf.c -o cups-pdf -lcups
+    runHook postBuild
+  '';
+  installPhase = ''
+    runHook preInstall
+    install -D --target-directory=${placeholder "out"}/lib/cups/backend cups-pdf
+    install -D --target-directory=${placeholder "out"}/etc/cups --mode=0644 cups-pdf.conf
+    install -D --target-directory=${placeholder "out"}/share/cups/model --mode=0644 *.ppd
+    runHook postInstall
+  '';
+  # without coreutils, cups-pdf would complain about missing `cp`
+  preFixup = ''
+    wrapProgram ${placeholder "out"}/lib/cups/backend/cups-pdf  \
+        --prefix PATH : "${lib.strings.makeBinPath [ coreutils ]}"
+  '';
+  passthru.tests.vmtest = nixosTests.cups-pdf;
+  meta = rec {
+    description = "A CUPS backend that turns print jobs into searchable PDF files";
+    downloadPage = homepage;
+    homepage = "https://github.com/alexivkin/CUPS-PDF-to-PDF";
+    license = lib.licenses.gpl2Only;
+    maintainers = [ lib.maintainers.yarny ];
+    longDescription = ''
+      cups-pdf is a CUPS backend that generates a PDF file for
+      each print job and puts this file into a folder on the local
+      machine such that the print job's owner can access the file.
+
+      https://www.cups-pdf.de/
+
+      ${pname} is a fork of cups-pdf which tries hard to preserve
```

```suggestion
      cups-pdf-to-pdf is a fork of cups-pdf which tries hard to preserve
```
substitutung pname wherever possible is not improving code quality